File: 03-triangle.rb
# Task: User able to get a right triangle's area result by inputting the width  and height value 

print "Please input the height of the right triangle, and hit Enter: "
a = gets.to_i

print "Please input the width of the right triangle, and hit Enter: "
b = gets.to_i

a *= a
b *= b

result = Math.sqrt(a + b)

puts "The area result is: #{result}"

File: 03-triangle/03-triangle.rs
use std::io;

fn read_float() -> f64 {
    let mut buf = String::new();
    io::stdin().read_line(&mut buf).unwrap();
    let f: f64 = buf.trim().parse().unwrap();
    f
}

fn main() -> io::Result<()> {
    println!("Please input the height of a right triangle:");
    let a = read_float();

    println!("Please input the width of a right triangle:");
    let b = read_float();

    let result = (a * a + b * b).sqrt();

    println!("The area of this square triangle is {}", result);

    Ok(())
}
